src/book/03-guides/04-general/01-learning-resources.mdx

```diff
@@ -0,0 +1,392 @@
+- [YouTube: CppCon](https://www.youtube.com/user/CppCon)
+
+**Tutorials:**
+
+- [NUbots Workshop: Modern C++](https://github.com/NUbots/NULearning/blob/master/cpp/moderncpp.md)
+
+</details>
+
+<details>
+<summary>NUClear</summary>
+
+The NUbots robot code uses the NUClear framework. Anyone developing robot code needs to learn how to work with NUClear. NUClear was created at NUbots as part of a Software Engineering FYP.
+
+**Reading:**
+
+- [NUbook: NUClear](/system/foundations/nuclear)
+
+**Videos:**
+
+- [YouTube: NUClear Basic Concepts](https://www.youtube.com/watch?v=Bx4yU-eA6bk&t=17m00s)
+- [YouTube: NUClear Example](https://www.youtube.com/watch?v=Bx4yU-eA6bk&t=32m00s)
+
+**Resources:**
+
+- [GitHub: NUClear](https://github.com/Fastcode/NUClear)
+- [GitHub: NUClear Roles](https://github.com/Fastcode/NUClearRoles)
+- [GitHub: NUClearNet](https://github.com/Fastcode/NUClearNet.js)
+- [GitHub: NUClearExample](https://github.com/Fastcode/NUClearExample)
+
+</details>
+
+<details>
+<summary>Robot Code</summary>
+
+**Reading:**
+
+- [NUbook: Getting Started](/guides/main/getting-started)
+- [NUbook: System Overview](/system/foundations/overview)
+- [NUbook: Troubleshooting](/guides/general/troubleshooting)
+- [NUbook: Glossary](/guides/general/glossary)
+
+**Resources:**
+
+- [GitHub: Robot Code](https://github.com/NUbots/NUbots)
+
+</details>
+
+<details>
+<summary> RoboCup Competition </summary>
+
+**Reading:**
+
+- [NUbook: RoboCup Resources](/team/robocup/resources)
+
+**Videos:**
+
+- [YouTube: Referee Training Video](https://www.youtube.com/watch?v=HHOtAklrcJU)
+- [YouTube: RoboCupSoccer Overview](https://www.youtube.com/watch?v=BN7t9dTbRyI)
+
+</details>
+
+<details>
+<summary> Using Webots </summary>
+
+The Webots simulator is used in some RoboCup competitions and is the simulator we use to aid in development of the robots.
+
+**Reading:**
+
+- [Setting up Webots](/guides/tools/webots-setup)
+- [Webots User Guide](https://cyberbotics.com/doc/guide/index)
+
+**Resources:**
+
+- [GitHub: NUWebots](https://github.com/NUbots/NUWebots)
+- [GitHub: Webots (Official)](https://github.com/cyberbotics/webots)
+- [GitHub: Webots (TC Version)](https://github.com/RoboCup-Humanoid-TC/webots)
+
+</details>
+
+<details>
+<summary> Mathematics </summary>
+
+Mathematics is integral part of any robotics system.
+
+**Reading:**
+
+- [NUbook: Mathematics](/system/foundations/mathematics#linear-algebra)
+- [Eigen C++ Linear Algebra Library](http://eigen.tuxfamily.org/dox/)
+
+**Videos:**
+
+- [3Blue1Brown: Essence of Linear Algebra](https://www.youtube.com/playlist?list=PLZHQObOWTQDPD3MizzM2xVFitgF8hE_ab)
+
+**Tutorials:**
+
+- [MIT Open Courseware: Linear Algebra](https://ocw.mit.edu/courses/mathematics/18-06-linear-algebra-spring-2010/video-lectures/)
+
+**Resources:**
+
+- [Wolfram Alpha](http://mathworld.wolfram.com/)
+
+</details>
+
+<details>
+<summary> Python </summary>
+
+Python is used in the build system in the main robot codebase.
```

Comment:
I wouldn't necessarily constrain python to the build system. Does interfacing with nbs files count as the build system? Does moving files onto a robot count as part of the build system? Does running binaries count as part of the build system? Some of the stuff we use python for is in a fairly grey area in this respect. I suggest saying it's used for managing code too? Idk

Reply:
'tools'?

Reply:
:shrug: sure?